source/slang/slang-serialize-ir.cpp

```diff
@@ -14,66 +17,299 @@
 //
 #include "slang-serialize-ir.cpp.fiddle"
 
+// If USE_RIFF is set, then we serialize using the RIFF backend, it's the
+// slowest option
+#define USE_RIFF 0
+// If we are serializing using Fossil, DIRECT_FROM_FOSSIL will make it so that
+// we unflatten directly from the fossilized representation rather than
+// deserializing everything first. It is the fastest option
+#define DIRECT_FROM_FOSSIL 0
+
 FIDDLE()
 namespace Slang
 {
 
+// The oldest compilers we support don't have std::bit_cast, so implement it
+// here
+template<typename To, typename From>
+    requires(sizeof(To) == sizeof(From)) && std::is_trivially_copyable_v<From> &&
+            std::is_trivially_copyable_v<To> && std::is_trivially_constructible_v<To>
+To bitCast(const From& src)
```

Comment:
we may want to move this to slang-common.h / slang-basic.h etc. so it is widely available.